src/averageAges.js

```diff
@@ -55,7 +66,25 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(person =>
+    (onlyWithSon ? person.sex === 'm' : true)
+    && people.find(mother => mother.name === person.mother)
+  );
+
+  const ageDifference = children.map(child => {
+    const personMother = people.find(mother => mother.name === child.mother);
+
+    return child.born - personMother.born;
+  });
+
+  return calculateAverageAge(ageDifference);
+}
+
+function calculateAverageAge(array) {
+  const yearsSum = array.reduce((sum, age) =>
+    sum + age, 0);
```

Comment:
you should use round braces if you move the return statement to the new line
```suggestion
  const yearsSum = array.reduce((sum, age) => (
    sum + age, 0)
  );
```